src/SDKs/Azure.ApplicationModel.Configuration/data-plane/README.md

```diff
@@ -41,43 +78,98 @@ Properties of a Configuration Setting:
     IDictionary<string, string> Tags { get; set; }
 ```
 
-# Hello World
-The following example demonstrates how to initialize a ConfigurationClient and perform a basic operations in the Configuration Store.
+## Examples
+The following sections provide several code snippets covering some of the most common Configuration Service tasks, including:
+- [Create a Configuration Setting](#create-a-Configuration-Setting)
+- [Retrieve a Configuration Setting](#retrieve-a-Configuration-Setting)
+- [Update an existing Configuration Setting](#update-an-existing-Configuration-Setting)
+- [Delete a Configuration Setting](#delete-a-Configuration-Setting)
 
-To begin using ConfigurationClient, a connection string must be provided which specifies the configuration store endpoint and credentials to use when sending requests. This conneection string can be retrieved by the Azure Portal or by using the [Azure CLI](https://docs.microsoft.com/en-us/azure/azure-app-configuration/cli-samples). From that point, the example uses the client to set, retrieve and delete a configuration setting by its name.
+### Create a Configuration Setting
+Create a Configuration Setting to be stored in the Configuration Store.
+There are two (2) ways to store a Configuration Setting:
+- AddAsync creates a setting only if the setting does not already exist in the store.
+- SetAsync creates a setting if it doesn't exist or overrides an existing setting.
 
 ```c#
-public async Task HelloWorld()
-{
-    // Connection string string from your Azure portal.
-    var connectionString = ...
+string connectionString = <connection_string>;
+var client = new ConfigurationClient(connectionString);
+var setting = new ConfigurationSetting("some_key", "some_value");
+await client.SetAsync(setting);
+```
 
-    // Instantiate a client that will be used to call the service.
-    var client = new ConfigurationClient(connectionString);
+### Retrieve a Configuration Setting
+Retrieves a previously stored Configuration Setting by calling GetAsync
 
-    // Create a setting to be stored by the configuration service.
-    var setting = new ConfigurationSetting("some_key", "some_value");
+```c#
+string connectionString = <connection_string>;
+var client = new ConfigurationClient(connectionString);
+var setting = new ConfigurationSetting("some_key", "some_value");
+await client.SetAsync(setting);
+ConfigurationSetting setting = await client.GetAsync("some_key");
+```
+
+### Update an existing Configuration Setting
+Updates an existing Configuration Setting by calling UpdateAsync. If the Configuration Setting does not exist in the store, a `412 - Precondition Failed` error will occur.
```

Comment:
Should we call out error conditions here? AddSetting can also response with a 412 if the key exists and GetSetting can response a 404 if the key doesn't exist, should those also be added?

I think this should be moved to the troubleshooting section.

Reply:
To avoid listing all the possible error codes that we can get, I added something like:

"When you interact with App Configuration using the .NET SDK, errors returned by the service correspond to the same HTTP status codes returned for [REST API][azconfig_rest] requests." 

like the sample readme does

Reply:
I was thinking that the sample section doesn't even mention errors, have the Troubleshooting section handle that.

Reply:
got it. removed